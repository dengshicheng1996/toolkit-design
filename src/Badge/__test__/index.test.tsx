import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Badge } from '..'

describe('Badge', () => {
  it('snapshot', () => {
    const component = shallow(
      <div className="demo-container">
        <h3>基础红点</h3>
        <div className="badge-container">
          <Badge dot />
          <Badge text={100}></Badge>
          <Badge text="99"></Badge>
          <Badge bubble placement="left" text="新气泡" />
        </div>
        <h3>数字红点</h3>
        <div className="badge-container">
          <Badge dot>
            <div className="number-cover" />
          </Badge>
          <Badge text={99}>
            <div className="number-cover" />
          </Badge>
          <Badge text={111}>
            <div className="number-cover" />
          </Badge>
        </div>
        <h3>文字红点</h3>
        <div className="badge-container">
          <Badge text="新">
            <div className="text-cover" />
          </Badge>

          <Badge text="NEW">
            <div className="text-cover" />
          </Badge>
          <Badge text="!">
            <div className="text-cover" />
          </Badge>
        </div>
        <h3>文字气泡</h3>
        <div className="badge-container">
          <Badge bubble placement="left" text="新气泡">
            <div className="bubble-cover" />
          </Badge>

          <Badge bubble placement="right" text="新气泡">
            <div className="bubble-cover" />
          </Badge>
          <Badge bubble placement="middle" text="新气泡">
            <div className="bubble-cover" />
          </Badge>
        </div>
      </div>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
