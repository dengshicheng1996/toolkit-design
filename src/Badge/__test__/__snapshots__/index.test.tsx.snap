// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge snapshot 1`] = `
<div
  className="demo-container"
>
  <h3>
    基础红点
  </h3>
  <div
    className="badge-container"
  >
    <Badge
      bubble={false}
      dot={true}
      overflowCount={99}
      placement="right"
      stroke={false}
    />
    <Badge
      bubble={false}
      dot={false}
      overflowCount={99}
      placement="right"
      stroke={false}
      text={100}
    />
    <Badge
      bubble={false}
      dot={false}
      overflowCount={99}
      placement="right"
      stroke={false}
      text="99"
    />
    <Badge
      bubble={true}
      dot={false}
      overflowCount={99}
      placement="left"
      stroke={false}
      text="新气泡"
    />
  </div>
  <h3>
    数字红点
  </h3>
  <div
    className="badge-container"
  >
    <Badge
      bubble={false}
      dot={true}
      overflowCount={99}
      placement="right"
      stroke={false}
    >
      <div
        className="number-cover"
      />
    </Badge>
    <Badge
      bubble={false}
      dot={false}
      overflowCount={99}
      placement="right"
      stroke={false}
      text={99}
    >
      <div
        className="number-cover"
      />
    </Badge>
    <Badge
      bubble={false}
      dot={false}
      overflowCount={99}
      placement="right"
      stroke={false}
      text={111}
    >
      <div
        className="number-cover"
      />
    </Badge>
  </div>
  <h3>
    文字红点
  </h3>
  <div
    className="badge-container"
  >
    <Badge
      bubble={false}
      dot={false}
      overflowCount={99}
      placement="right"
      stroke={false}
      text="新"
    >
      <div
        className="text-cover"
      />
    </Badge>
    <Badge
      bubble={false}
      dot={false}
      overflowCount={99}
      placement="right"
      stroke={false}
      text="NEW"
    >
      <div
        className="text-cover"
      />
    </Badge>
    <Badge
      bubble={false}
      dot={false}
      overflowCount={99}
      placement="right"
      stroke={false}
      text="!"
    >
      <div
        className="text-cover"
      />
    </Badge>
  </div>
  <h3>
    文字气泡
  </h3>
  <div
    className="badge-container"
  >
    <Badge
      bubble={true}
      dot={false}
      overflowCount={99}
      placement="left"
      stroke={false}
      text="新气泡"
    >
      <div
        className="bubble-cover"
      />
    </Badge>
    <Badge
      bubble={true}
      dot={false}
      overflowCount={99}
      placement="right"
      stroke={false}
      text="新气泡"
    >
      <div
        className="bubble-cover"
      />
    </Badge>
    <Badge
      bubble={true}
      dot={false}
      overflowCount={99}
      placement="middle"
      stroke={false}
      text="新气泡"
    >
      <div
        className="bubble-cover"
      />
    </Badge>
  </div>
</div>
`;
